tests/auth_basic_test.py

```diff
@@ -0,0 +1,189 @@
+    )
+    assert response.status_code == 200
+    assert len(capture_templates) == 1
+    assert capture_templates[0][0].name == "home.html", "Wrong template used"
+
+
+def test_signup_post_weak_password(client, capture_templates):
+    """Tests the signup functionality with a weak password.
+
+    Verify that the system enforces strong password requirements.
+    """
+    response = client.post(
+        "/signup",
+        data={"email": "new_user@example.com", "password": "weak"},
+        follow_redirects=True,
+    )
+    assert response.status_code == 200
+    assert b"Please enter a valid password" in response.data
+    assert len(capture_templates) == 1
+    assert capture_templates[0][0].name == "signup.html", "Wrong template used"
+
+
+def test_login_route(client, capture_templates):
+    """Tests the accessibility of the login route.
+
+    Ensure the login page is accessible and rendered correctly.
+    """
+    response = client.get("/login")
+    assert response.status_code == 200
+    assert b"Login" in response.data
+    assert len(capture_templates) == 1
+    assert capture_templates[0][0].name == "login.html", "Wrong template used"
+
+
+def test_login_valid_credentials(client, capture_templates, test_user):
+    """Tests login functionality with valid credentials.
+
+    Ensure that users can log in successfully and are redirected to their
+    profile page.
+    """
+    response = client.post(
+        "/login",
+        data={"email": "test@example.com", "password": "TestP@ssword!"},
+        follow_redirects=False,
+    )
+    response = client.get(
+        response.headers.get("Location"), follow_redirects=True
+    )
+    assert response.status_code == 200
+    assert len(capture_templates) == 1
+    assert capture_templates[0][0].name == "profile.html", "Wrong template used"
+
+
+def test_login_invalid_credentials(client, capture_templates):
+    """Tests login functionality.
+
+    Tests login functionality with invalid credentials to confirm system
+    correctly identifies incorrect login attempts and prevents access.
+    """
+    response = client.post(
+        "/login",
+        data={"email": "test@example.com", "password": "wrongpassword"},
+        follow_redirects=True,
+    )
+    assert response.status_code == 200
+    assert b"Please check your login details and try again." in response.data
+    assert len(capture_templates) == 1
+    assert capture_templates[0][0].name == "login.html", "Wrong template used"
+
+
+def test_logout(client, capture_templates, test_user, login_client):
+    """Tests the logout functionality.
+
+    Tests the logout functionality to verify that a logged-in user can
+    successfully log out and is redirected to the home page.
+    """
+    response = client.get("/logout", follow_redirects=True)
+    assert response.status_code == 200
+    assert len(capture_templates) == 2
+    assert capture_templates[1][0].name == "home.html", "Wrong template used"
```

Comment:
Could you add in a check here that tries to access a page that is `@login_required` after you logout? Fine if you use/edit this.

```
profile_response = client.get("/profile", follow_redirects=True)
assert profile_response.status_code == 200, "Response after redirection"
assert b"Login" in profile_response.data, "User prompted to log in"
assert "login.html" in [
    template[0].name for template in capture_templates
], "Redirect to login page after trying to access profile post-logout"
```